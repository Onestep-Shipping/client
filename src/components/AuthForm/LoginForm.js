import React, {useState, useCallback} from 'react';
import './AuthForm.css';

import Input from '../Input/Input';
import {useHistory} from 'react-router-dom';
import app from '../../firebase/base.js';

const LoginForm = () => {
  const history = useHistory();

  const [errors, setErrors] = useState([]);
  const [displayErrors, setDisplayErrors] = useState(false);

  const handleLogin = useCallback(async (e) => {
    e.preventDefault();
    const email = e.target[0].value;
    const password = e.target[1].value;
    try {
      await app
          .auth()
          .signInWithEmailAndPassword(email, password);
      history.goBack();
    } catch (error) {
      alert(error);
    }
  }, [history]);

  return (
    <div className="inner-container">
      <div className="header">Login</div>
      <form className="box" onSubmit={handleLogin} noValidate>
        {errors.length > 0 &&
          errors.map((err, ind) => {
            return <small className="danger-error" key={ind}>{err}</small>;
          })
        }
        <Input name="Email" type="email" displayErrors={displayErrors} />

        <Input
          name="Password"
          type="password"
          displayErrors={displayErrors}
        />

        <button className="login-btn">Login</button>
      </form>
    </div>
  );
};

export default LoginForm;
