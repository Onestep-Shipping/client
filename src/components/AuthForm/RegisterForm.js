/* eslint-disable no-unused-vars */
import React, {useState, useCallback} from 'react';
import './AuthForm.css';

import Input from '../Input/Input';
import app from '../../firebase/base.js';
import PropTypes from 'prop-types';

const RegisterForm = (props) => {
  const { closeModal } = props;

  const [errors, setErrors] = useState([]);
  const [displayErrors, setDisplayErrors] = useState(false);


  const handleSignUp = useCallback(async (e) => {
    e.preventDefault();
    const email = e.target[6].value;
    const password = e.target[7].value;
    if (email.length > 0 && password.length > 0) {
      try {
        await app
            .auth()
            .createUserWithEmailAndPassword(email, password);
        closeModal();
      } catch (error) {
        alert(error);
      }
    }
  }, [closeModal]);

  return (
    <div className="inner-container">
      <div className="header">Register</div>
      <form className="register-box" onSubmit={handleSignUp} noValidate>
        {errors.length > 0 &&
          errors.map((err, ind) => <small className="danger-error" key={ind}>{err}</small>)
        }
        <div className="all-texfield-register-container">
          <div className="company-info-container">
            <Input name="Company Name" type="text" displayErrors={displayErrors} />
            <Input name="Address" type="address" 
              displayErrors={displayErrors} 
              placeholder="Street"
            />
            <Input name="" type="address" 
              displayErrors={displayErrors} 
              placeholder="City, Province & Postal Code"
            />
            <Input name="" type="address" 
              displayErrors={displayErrors} 
              placeholder="Country"
            />
            <Input name="Telephone" type="phone" displayErrors={displayErrors} />
          </div>
          <div>
            <Input name="Tax ID" type="text" displayErrors={displayErrors} />
            <Input name="Person In-Charge" type="text" displayErrors={displayErrors} />
            <Input name="Position" type="text" displayErrors={displayErrors} />
            <Input name="Email" type="email" displayErrors={displayErrors} />
            <Input name="Password" type="password" displayErrors={displayErrors} />
          </div>
        </div>
        <button className="login-btn">Sign Up</button>
      </form>
    </div>
  );
};

export default RegisterForm;

RegisterForm.propTypes = {
  closeModal: PropTypes.func,
};
