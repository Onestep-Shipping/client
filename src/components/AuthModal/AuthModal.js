import React from 'react';
import Modal from 'react-modal';
import LoginForm from '../AuthForm/LoginForm.js';
import RegisterForm from '../AuthForm/RegisterForm.js';
import PropTypes from 'prop-types';

const customStyles = {
  content : {
    top                   : '50%',
    left                  : '50%',
    right                 : 'auto',
    bottom                : 'auto',
    transform             : 'translate(-50%, -50%)',
    borderRadius          : '10px',
    padding               : '0px',
    border                : 'none'
  }
};

const AuthModal = (props) => {
  const { isModalOpen, closeModal, type } = props;

  return (
    <Modal
      isOpen={isModalOpen}
      onRequestClose={closeModal}
      style={customStyles}
    >
      {type === 'login' ? 
        <LoginForm closeModal={closeModal}/> : 
        <RegisterForm closeModal={closeModal}/>}
    </Modal>
  );
}

export default AuthModal;

AuthModal.propTypes = {
  isModalOpen: PropTypes.bool,
  closeModal: PropTypes.func,
  type: PropTypes.string,
};
