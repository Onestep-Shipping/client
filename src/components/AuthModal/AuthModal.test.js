import AuthModal from './AuthModal';
import LoginForm from '../AuthForm/LoginForm.js';
import React from 'react';
import RegisterForm from '../AuthForm/RegisterForm.js';
import { shallow } from 'enzyme';

const loginType = 'login';
const registerType = 'register';
const isModalOpen = false;
const closeModal = () => {}

let wrapper = shallow( 
  <AuthModal
    isModalOpen={isModalOpen}
    closeModal={closeModal}
    type={loginType} />
);

describe('AuthModal', () => {
  it('should render the AuthModal Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
  it('should contains Login Form if type is login', () => { 
    expect(wrapper).toContainReact(
      <LoginForm closeModal={closeModal} />
    );
  });
  it('should contains Register Form if type is not login', () => { 
    wrapper.setProps({ type: registerType });
    expect(wrapper).toContainReact(
      <RegisterForm closeModal={closeModal} />
    );
  });
});