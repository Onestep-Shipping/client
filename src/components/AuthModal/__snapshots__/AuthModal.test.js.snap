// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthModal should render the AuthModal Component correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "border": "none",
        "borderRadius": "10px",
        "bottom": "auto",
        "left": "50%",
        "padding": "0px",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      },
    }
  }
>
  <LoginForm
    closeModal={[Function]}
  />
</Modal>
`;
